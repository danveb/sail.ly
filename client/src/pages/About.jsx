import { aboutSailing } from "../constants/about"; 
import aboutSailTop from "../assets/about/about.jpg"; 
import "../styles/About.css"; 

export const About = () => {
    return (
        <div className="about">
            <div className="about__wrapper">
                <div className="about__main">
                    <div className="about__top">
                        <img src={aboutSailTop} alt="sailing yacht in ocean" />
                    </div>
                    <div className="about__bottom">
                        {aboutSailing.map((item) => (
                            <div key={item.id}>
                                <h2>{item.header}</h2>
                                <p>{item.text_intro}</p>
                                <p>{item.text_main}</p>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        </div>
    )
}
