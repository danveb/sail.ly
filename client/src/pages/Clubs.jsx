// import { ClubList } from "../components/ClubList";
import { clubsIntro } from "../constants/clubs";
import clubIntro from "../assets/clubs/club-intro.jpg"; 
import "../styles/Clubs.css"; 
import { ClubList } from "../components/ClubList";

export const Clubs = () => {
    return (
        <>
            <div className="clubs">
                <div className="clubs__wrapper">
                    <div className="clubs__main">
                        <div className="clubs__top">
                            <img src={clubIntro} alt="sailing yacht in ocean" />
                        </div>
                        <div className="clubs__bottom">
                            {clubsIntro.map((item) => (
                                <div key={item.id}>
                                    <h2>{item.header}</h2>
                                    <p>{item.text_intro}</p>
                                    <p>{item.text_main}</p>
                                </div>
                            ))}
                        </div>
                    </div>
                </div>
            </div>
            <ClubList />
        </>
    )
}