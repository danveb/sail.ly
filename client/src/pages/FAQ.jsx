import { FAQItem } from "../components/FAQItem";
import faq from "../assets/faq/faq.jpg"; 
import "../styles/FAQ.css"; 

export const FAQ = () => {
    return (
        <div className="faq">
            <div className="faq__wrapper">
                <div className="faq__main">
                    <div className="faq__top">
                        <img src={faq} alt="sailing yacht in ocean" />
                    </div>
                    <div className="faq__bottom">
                        <FAQItem />
                    </div>
                </div>
            </div>
        </div>
    )
}