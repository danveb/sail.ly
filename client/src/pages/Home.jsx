import chevronDown from "../assets/home/chevron-down.svg"; 
import "../styles/Home.css"; 

export const Home = () => {
    return (
        <div>
            <div className="home">
                <div className="home__top">
                    <div className="home__top--title">
                        <h1>#SS | 23</h1>
                        <h3>Dream in open waters</h3>
                        {/* LINK to clubs */}
                    </div>
                    <div className="home__top--bg"></div>
                    <img src={chevronDown} alt="chevron down arrow" />
                </div>
            </div>
        </div>
    )
}