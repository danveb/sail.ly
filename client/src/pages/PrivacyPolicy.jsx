import "../styles/PrivacyPolicy.css"; 

export const PrivacyPolicy = () => {
    return (
        <div className="privacy">
            <div className="privacy__wrapper">
                <div className="privacy__main">
                    <div className="privacy__top" data-testid="privacy__top">
                        <h1>Privacy Policy</h1>
                        <p>At SAIL.LY we take your privacy seriously. This Privacy Policy explains how we collect, use, and disclose your personal information when you use our web app.</p>
                    </div>
                    <div className="privacy__bottom">
                        <h2>Information We Collect</h2>
                        <p>When you use SAIL.LY we do not collect any of the following personal information:</p>
                        <ul>
                            <li>Contact Information: Name and email address</li>
                            <li>Usage Information: Information about how you use our web app, such as the pages you visit and the features you use</li>
                            <li>Device Information: Information about the device you use to access our web app, such as your IP address, browser type, and operating system</li>
                        </ul>
                        <h2>How We Use Your Information</h2>
                        <p>We do not use your personal information to provide and/or improve our web app. This app is built and designed for anyone that may want to learn more about climatic conditions pertaining to sailing.</p>
                        <h2>Data Retention</h2>
                        <p>We do not retain any personal information simply because it is not needed.</p>
                        <h2>Changes to This Privacy Policy</h2>
                        <p>We may update this Privacy Policy if needed. If we make any material changes we will post a notice on our web app prior to the effective date of the changes.</p>
                    </div>
                </div>
            </div>
        </div>
    )
}