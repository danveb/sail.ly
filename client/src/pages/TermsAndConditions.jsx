import "../styles/TermsAndConditions.css"; 

export const TermsAndConditions = () => {
    return (
        <div className="terms" data-testid="terms">
            <div className="terms__wrapper">
                <div className="terms__main">
                    <div className="terms__top">
                        <h1>Terms And Conditions</h1>
                        <p>Welcome to SAIL.LY! Please read these Terms and Conditions carefully before using our web app. By accessing or using SAIL.LY, you agree to be bound by these Terms and Conditions. If you do not agree to these Terms and Conditions, please do not use our web app.</p>
                    </div>
                    <div className="terms__bottom">
                        <h2>Use of SAIL.LY</h2>
                        <p>SAIL.LY is an application intended for educational purposes. It serves anyone who has little-to-no experience up to experienced sailors.</p>
                        <h2>Intellectual Property</h2>
                        <p>You agree that all materials, products, and services provided on this payment application are the property of SAIL.LY, its affiliates, directors, officers, employees, agents, suppliers, or licensors including all copyrights, trade secrets, trademarks, patents, and other intellectual property.</p>
                        <h2>Applicable Law</h2>
                        <p>By visiting this website, you agree that the laws of the United States of America, without regard to principles of conflict laws, will govern these terms and conditions, or any dispute of any sort that might come.</p>
                    </div>
                </div>
            </div>
        </div>
    )
}