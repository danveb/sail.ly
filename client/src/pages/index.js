export { default as About } from "./About";
export { default as Clubs } from "./Clubs";
export { default as FAQ } from "./FAQ";
export { default as Home } from "./Home";
export { default as PrivacyPolicy } from "./PrivacyPolicy";
export { default as TermsAndConditions } from "./TermsAndConditions"; 