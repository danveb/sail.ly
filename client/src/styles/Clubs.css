.clubs {
    background-color: #fff; 
}

.clubs__wrapper {
    max-width: 1200px; 
    height: 100%; 
    margin: 0 auto; 
}

.clubs__main {
    display: flex; 
    flex-direction: column; 
    justify-content: center; 
    align-items: center; 
    max-width: 590px; 
    margin: auto; 
}

.clubs__top {
    display: flex; 
}

.clubs__top img {
    width: 100%; 
    height: 100%; 
    object-fit: contain; 
}

.clubs__bottom {
    margin: 30px 0px; 
}

@media screen and (max-width: 591px) {
    .clubs__bottom {
        margin: auto; 
        padding: 25px; 
    }
}

.clubs__bottom h2 {
    font-size: 24px; 
    font-family: "Prata", sans-serif; 
    margin: 15px 0; 
}

.clubs__bottom p {
    font-family: "Yrsa", sans-serif; 
    color: #333; 
    margin: 15px 0; 
}

.clubList__wrapper {
    max-width: 1200px; 
    margin: 0 auto; 
    width: 100%;
    height: 100%; 
    display: flex; 
    flex-direction: column; 
}

.clubList__search {
    width: 100%; 
    height: 100%; 
    display: flex; 
    align-items: center; 
    justify-content: center; 
    margin: 15px 0; 
    position: relative; 
}

.clubList__search input {
    padding: 10px 15px; 
    font-family: "Yrsa", sans-serif; 
    font-size: 13px; 
    border-radius: 50px; 
    border-width: 0.5px; 
    width: 50%; 
}

.clubList__search img {
    width: 15px; 
    height: 15px; 
    position: absolute; 
    right: 26%; 
}

.clubList__grid {
    display: grid; 
    grid-template-columns: repeat(3, minmax(0, 1fr)); 
    gap: 0.5rem; 
    margin-bottom: 30px; 
}

@media screen and (max-width: 576px) {
    .clubList__grid {
        grid-template-columns: repeat(2, minmax(0, 1fr));
    }
}

@media screen and (max-width: 480px) {
    .clubList__grid {
        grid-template-columns: repeat(1, minmax(0, 1fr));
    }
}