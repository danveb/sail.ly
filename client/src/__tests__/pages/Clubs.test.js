import { render, screen } from "@testing-library/react";
import { Clubs } from "../../pages/Clubs"; 

describe("Clubs Component", () => {
    // smoke test
    test("renders without crashing", () => {
        render(<Clubs />); 
    });

    // snapshot test 
    // "cannot log after tests are done... axios error?"
    test("matches snapshot", () => {
        const { asFragment } = render(<Clubs />); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByPlaceholderText
    // test("renders the correct placeholder", () => {
    //     const placeholderText = screen.getByPlaceholderText("Search Club by Name or City"); 
    //     expect(placeholderText).toBeInTheDocument(); 
    // });

    // getByAltText 
    test("img tag has attribute for screen readers", () => {
        render(<Clubs />); 
        const altText = screen.getByAltText("sailing yacht in ocean"); 
        expect(altText).toBeInTheDocument(); 
    });

    // getByText h2 
    test("h2 tag correctly renders text", () => {
        render(<Clubs />); 
        const textElement = screen.getByText("Sailing Clubs"); 
        expect(textElement).toBeInTheDocument(); 
        expect(textElement).not.toBe("Clubs Sailing"); 
    });

    // getByText p
    test("p tag renders text-intro", () => {
        render(<Clubs />); 
        const textElement = screen.getByText("Visit your nearest sailing club to learn more about sailing resources, support, camaraderie and events to become part of the sailing community.", { exact: true }); 
        expect(textElement).toBeInTheDocument(); 
    });
}); 