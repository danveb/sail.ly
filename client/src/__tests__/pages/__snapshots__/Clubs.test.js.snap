// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clubs Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="clubs"
  >
    <div
      class="clubs__wrapper"
    >
      <div
        class="clubs__main"
      >
        <div
          class="clubs__top"
        >
          <img
            alt="sailing yacht in ocean"
            src="club-intro.jpg"
          />
        </div>
        <div
          class="clubs__bottom"
        >
          <div>
            <h2>
              Sailing Clubs
            </h2>
            <p>
              Sailing clubs offer sailing beginners and enthusiasts a range of facilities and services. Members come from all walks of life, with varying levels of experience and skill in sailing.
            </p>
            <p>
              Visit your nearest sailing club to learn more about sailing resources, support, camaraderie and events to become part of the sailing community.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="clubList"
  >
    <div
      class="clubList__wrapper"
    >
      <form
        class="clubList__search"
      >
        <input
          id="search"
          placeholder="Search Club by Name or City"
          type="text"
        />
        <img
          alt="search logo"
          src="search.svg"
        />
      </form>
      <div
        class="clubList__grid"
      />
    </div>
  </div>
</DocumentFragment>
`;
